gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py

```
@@ -848,10 +848,18 @@ def stop_segments(context, segment_type):
     for seg in segments:
         # For demo_cluster tests that run on the CI gives the error 'bash: pg_ctl: command not found'
         # Thus, need to add pg_ctl to the path when ssh'ing to a demo cluster.
-        subprocess.check_call(['ssh', seg.getSegmentHostName(),
-                               'source %s/greenplum_path.sh && pg_ctl stop -m fast -D %s' % (
-                                   pipes.quote(os.environ.get("GPHOME")), pipes.quote(seg.getSegmentDataDirectory()))
-                               ])
+        try:
+            subprocess.check_call(['ssh', seg.getSegmentHostName(),
+                                   'source %s/greenplum_path.sh && pg_ctl stop -m '
+                                   'fast -D %s -w' % (
+                                       pipes.quote(os.environ.get("GPHOME")),
+                                       pipes.quote(seg.getSegmentDataDirectory()))
+                                  ])
+        except Exception as e:
+            raise Exception("pg_ctl stop for host %s and data directory %s "
+                            "failed with error %s"
+                            %(seg.getSegmentHostName(),
+                            seg.getSegmentDataDirectory, e.message))
```

I think you want to make this a function call: `seg.getSegmentDataDirectory()`

---

This routine will fail-fast on the first segment that didn't stop within 60 seconds.  Is this the semantic we want?   I was thinking we will get more information from trying to stop all the segments.  On the other hand, we might not want to wait N times if there is something wrong with the cluster.

---

We have changed to make the call `seg.getSegmentDataDirectory()`.

We have not changed any waiting behavior or added fail fast. Previously the exception was just passed to the calling layer. We have just intercepted it to print out the host and data directory information and continue to pass it to the caller as before.